### ISSUE TITLE
[HOLD for payment 2024-06-28] [MEDIUM] Create Trip Room Preview

### ISSUE THREAD

**Issue by stitesExpensify** [Expensify/App#37826]
## VIP-Travel Implementation Issue
Tracking Issue: https://github.com/Expensify/Expensify/issues/333471
Project: https://github.com/orgs/Expensify/projects/112/
Relevant Design Doc Section: https://docs.google.com/document/d/122DLLTuCsU4NC3c7rNiy4UF8bZxVlLEWSy3yncqCTcQ/edit#bookmark=id.6c8m84z62q6z

This issue is for creating another new preview type, this time for trip rooms:
<img width="807" alt="2024-03-06_08-39-00" src="https://github.com/Expensify/App/assets/42391420/2adc15e1-1fc3-471f-a367-93f677f072eb">

- A new component, TripRoomPreview will be created in src/components/ReportActionItem.
- A new action type, ACTION_TRIPPREVIEW, will be added to [CONST.ts](https://github.com/Expensify/App/blob/d5eba656251269b238797817df3cf84800b0039a/src/CONST.ts#L525).
- If the report action is of the new type, which we will check in [ReportActionItem.js](https://github.com/Expensify/App/blob/d5eba656251269b238797817df3cf84800b0039a/src/pages/home/report/ReportActionItem.js#L397), the new component will be displayed.
- TripRoomPreview will mimic [ReportPreview](https://github.com/Expensify/App/blob/main/src/components/ReportActionItem/ReportPreview.tsx)’s formatting and behavior with some notable changes:
	- We’ll display the room name under the total
- [Receipts](https://github.com/Expensify/App/blob/d5eba656251269b238797817df3cf84800b0039a/src/components/ReportActionItem/ReportPreview.tsx#L249) will not be displayed.
- Instead, a new helper function, getTripTransactions(expenseReportID) will be created in [ReportUtils](https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.ts), and will return any transaction with travel details. We’ll display those details as shown in the mockup.
- On click, it’ll take the user to the trip room via Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(reportID));

<details><summary>Issue Owner</summary>Current Issue Owner: @twisterdotcom</details>

**Comment by WojtekBoman** [comment]
Hey! I’m Wojtek from Software Mansion, the expert agency, and I’d like to work on this issue.

**Comment by twisterdotcom** [comment]
Adding a C+ here for Travel reviews: @shubham1206agra

**Comment by twisterdotcom** [comment]
Only assigned on Friday. We can give them a day here Melvin.

**Comment by twisterdotcom** [comment]
Are we waiting on anything internal here?

**Comment by WojtekBoman** [comment]
Sorry I was OOO, I'm currently waiting for a response from @stitesExpensify [here](https://swmansion.slack.com/archives/C05S5EV2JTX/p1710773674911979?thread_ts=1710235822.680919&cid=C05S5EV2JTX)

**Comment by WojtekBoman** [comment]
@shawnborton How do we want to format the trip duration text when the start and end dates are in different months/years?
![Screenshot 2024-03-19 at 14 41 54](https://github.com/Expensify/App/assets/47774969/c1f2bf83-c7a4-46b3-87be-125cf4e730b7)

**Comment by shawnborton** [comment]
Good question. Maybe we should standardize on something like:

- Mar 17-20
- Feb 28 to March 1
- Dec 28, 2023 to Jan 5, 2024

When we do timestamps on messages in the product, we do the month abbreviation (Jan, Feb, Mar, etc) so I think we should stay consistent here too. 

cc @Expensify/design @twisterdotcom @ryanschaffer for thoughts on that

**Comment by dannymcclain** [comment]
I like the formats you've laid out.

**Comment by twisterdotcom** [comment]
Agree with this. Why would we do:

>Feb 28 to March 1

And not:

>Feb 28 to Mar 1

**Comment by shawnborton** [comment]
Totally agree, that was a typo on my part! Will update my comment.

**Comment by WojtekBoman** [comment]
@twisterdotcom I see in the Spotnana documentation that we can also book a rail travel or a limo. Spotnana also provides trip type: miscellaneous. Do we want to handle these types of trips as well?
<img width="659" alt="Screenshot 2024-03-21 at 17 09 09" src="https://github.com/Expensify/App/assets/47774969/b0e85eec-e78d-4ce9-b3bb-bec453bf5128">

**Comment by WojtekBoman** [comment]
Yes, I think the best solution for us would be to have separate types for them and display the appropriate icons based on that

**Comment by twisterdotcom** [comment]
Okay, there won't be any limos, that's a specific paid VIP extra we're not handling yet.

Misc might happen, can be random things like ferries which sometimes are booked with rail travel tickets etc. So, we could just use the luggage icon for those, right @shawnborton / @dannymcclain?

**Comment by dannymcclain** [comment]
@twisterdotcom Yeah that's probably what I'd do. I went ahead and mapped some icons from our system to these categories (potential alternates in the light gray)

![image](https://github.com/Expensify/App/assets/7516624/9fe03896-121a-4beb-935c-9341e437050d)

🙏 please don't make me make a limo icon 🤣

**Comment by dannymcclain** [comment]
Oh I'm just now seeing that Shawn already laid out some icons for these different categories. My bad. Feel free to ignore me and proceed!

**Comment by WojtekBoman** [comment]
@twisterdotcom I'll prepare today the draft PR for this issue :) However, I'd like to know if we will be able to adjust the created components when the backend is ready? We currently work mainly on dummy data and I'd like to test whether everything works fine with real data obtained from our backend.

**Comment by twisterdotcom** [comment]
@WojtekBoman yes, we can make adjustments once we get production data of course.

**Comment by WojtekBoman** [comment]
@twisterdotcom @dannymcclain @shawnborton I'd like to confirm that you accept these designs for new trip types, so I'm attaching screenshots here :) Also, I wonder for which types of trip exactly we should display the arrow, on the designs it was only used for flights, but I think it could be displayed also for trains, what do you think about it?

<img width="794" alt="Screenshot 2024-03-22 at 13 43 03" src="https://github.com/Expensify/App/assets/47774969/0f256e71-8da0-49f1-a9e5-de86c031ded2">

<img width="856" alt="Screenshot 2024-03-22 at 13 41 34" src="https://github.com/Expensify/App/assets/47774969/5617f017-6a44-4a8a-961b-d784633a546c">

**Comment by dannymcclain** [comment]
@WojtekBoman That's looking pretty good to me, although Shawn was the one who originally designed this so I'll let him weigh in too.

For the use of the arrow, I think it could make sense to use that for train/rail as well. Basically anything that has a set and defined start and end point could use the arrows in my mind? But if we do want to use it for trains, we'll have to think of a graceful way to truncate the start/end addresses.

Also, small thing, but can we add this updated luggage icon to our assets and use it in place of the one you're currently using? [Luggage with Lines.svg.zip](https://github.com/Expensify/App/files/14723015/Luggage.with.Lines.svg.zip)

> [!NOTE]
> I don't think we should _replace_ the current luggage icon in the repo—I think we should just add the new one so we have both. Let me know if anyone feels strongly against that approach.

**Comment by dubielzyk-expensify** [comment]
> I don't think we should replace the current luggage icon in the repo—I think we should just add the new one so we have both. Let me know if anyone feels strongly against that approach.

I was originally with you, but when thinking about it I just wonder if it creates more inconsistency with them both existing. We added the new one for a reason and I think that still applies, so while I prefer the older travel icon, I kinda lean on replacing it so we end up with one travel icon across the app.

**Comment by shawnborton** [comment]
Yeah, I think I lean towards only having one luggage icon in the repo given that it's the only one we plan on using, mostly to prevent any confusion. 

The mock above looks good though. Same comment as Danny, would be nice to repeat the arrows for trains or maybe we should only ever do "SFO to JFK" so that there isn't an outlier here with arrows whereas others do not.

**Comment by dannymcclain** [comment]
Ok cool that works for me re: replacing the luggage icon! Also after thinking about it more you 2 are definitely right.

**Comment by WojtekBoman** [comment]
@twisterdotcom 

I'd like to ask about sending PRs related to the travel module for review. How should we handle that? During our work, we mainly based on dummy data and currently it's not possible to test it without making some changes in code (preparing data for tests etc.). We've also added comments with the annotation `@TODO` in places in code where we aren't sure about the data structure, we'll remove these comments once we compare types defined by us with the data obtained from the API. So, do we want to send these PRs now for review, and then create another PR with adjusting components when the backend is ready, or should we wait for the backend and after that make them ready for review? 

These features are of course available only for users who have access to the travel beta. However, problems may occur if the backend is ready and users get access to the beta, and we don't adapt the components first.

**Comment by stitesExpensify** [comment]
Personally I think that we should wait. As long as everything is ready to go besides being connected to the backend, then it should really make a difference timing-wise. 

My concern is that if we merge it, then we will get a flood of bug reports from contributors because all contributors are on all betas in dev

**Comment by twisterdotcom** [comment]
Just waiting on webhooks

**Comment by twisterdotcom** [comment]
Still waiting on webhooks.

**Comment by twisterdotcom** [comment]
Waiting on the backend.

**Comment by twisterdotcom** [comment]
Waiting on the backend still

**Comment by twisterdotcom** [comment]
Spotnana thread for backend updates here: https://expensify.slack.com/archives/C066REXJ9D3/p1713220956933639

**Comment by twisterdotcom** [comment]
Updates in https://expensify.slack.com/archives/C05S5EV2JTX/p1714085516706139?thread_ts=1713825695.223409&cid=C05S5EV2JTX

**Comment by twisterdotcom** [comment]
Update here: https://expensify.slack.com/archives/C05S5EV2JTX/p1714431147318139. Waiting on Spotnana.

**Comment by rushatgabhane** [comment]
@twisterdotcom nothing yet. I'm prioritizing https://github.com/Expensify/App/pull/41659 first

**Comment by smelaa** [comment]
Hi @rushatgabhane, I am Aleksandra Smela from Software Mansion and since we are back in the office after a holiday we can continue working on this issue in case you haven't started yet. 
If you would like a hand with this let me know if the backend is ready - that was the blocker of [Wojtek's PR](https://github.com/Expensify/App/pull/38808).

**Comment by rushatgabhane** [comment]
@smelaa yes please feel free to continue working on this

**Comment by smelaa** [comment]
@stitesExpensify since Ted is OOO could you please assign me and give an update on how the work on backend is going?

**Comment by stitesExpensify** [comment]
The backend is completely ready to go but we have a merge freeze. We are plan on getting an Auth deploy out tonight which will start sending trip reports and transactions to NewDot (without the trip name and date) so hopefully you can get real data starting to build off of tomorrow

**Comment by smelaa** [comment]
Working on this since the deploy is out.

**Comment by twisterdotcom** [comment]
Looks like we rolled back some updates for the webhooks so this is on hold again.

**Comment by twisterdotcom** [comment]
Hopefully we'll deploy tomorrow/weds.

**Comment by twisterdotcom** [comment]
This is off hold now! You should be able to get moving on it now.

**Comment by twisterdotcom** [comment]
Do we have a PR ongoing now?

**Comment by WojtekBoman** [comment]
> Do we have a PR ongoing now?

Yes https://github.com/Expensify/App/pull/38808. I see that trip data is already available in NewDot, so I'll continue working on that today

**Comment by twisterdotcom** [comment]
Payment Summary:

- @WojtekBoman **does not** require payment (Contractor)
- @shubham1206agra paid $250 [here](https://www.upwork.com/nx/wm/offer/102911935)
- @smelaa **does not** require payment (Contractor)

**Comment by twisterdotcom** [comment]
Thanks! Just updated it.
